ext/bg/js/backend.js

```diff
@@ -315,6 +314,61 @@ class Backend {
         return await this.dictionaryImporter.import(this.database, archiveSource, onProgress, details);
     }
 
+    async _textParseScanning(text, options) {
+        const results = [];
+        while (text.length > 0) {
+            const term = [];
+            const [definitions, sourceLength] = await this.translator.findTerms(
+                'simple',
+                text.substring(0, options.scanning.length),
+                {},
+                options
+            );
+            if (definitions.length > 0) {
```

Comment:
`if (definitions.length > 0 && sourceLength > 0) {`

Explicitly prevent infinite loops. I _hope_ this case should never happen, but better safe than sorry.

Reply:
https://github.com/FooSoft/yomichan/pull/456/commits/3fa8b8e1866933fdfccd0598a11e443fc1693840